src/pixel/state.ts

```diff
@@ -50,7 +51,12 @@ const paramExtractors: ((state: State) => [string, string][])[] = [
   ifDefined('privacyMode', privacyMode => asStringParamWhen('nb', privacyMode ? 1 : 0, v => v === 1)),
   ifDefined('gdprConsent', gdprConsentString => asStringParam('gdpr_consent', gdprConsentString)),
   ifDefined('referrer', referrer => asStringParam('refr', referrer)),
-  ifDefined('contextElements', contextElements => asStringParam('c', contextElements))
+  ifDefined('contextElements', contextElements => asStringParam('c', contextElements)),
+  state => asStringParamWhen('pu_rp', isPageRemoved(state.pageUrl, state) ? 1 : 0, v => v === 1),
+  state => {
+    const blockedParams = blockedQueryParams(state.pageUrl, state)
```

Comment:
There are currently multiple places at which the URL is parsed and query parameters are filtered. Would it be possible to do only one single pass for all of these steps: if the config requires adjusting the original URL, parse it and do one pass to return the adjusted URL with the info what has been adjusted - e.g. page removed flag, removed query parameters list.